cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -129,9 +130,23 @@ public static void createBalaInHomeRepo(Response balaDownloadResponse, Path pkgP
                     logFormatter.formatLog("error accessing bala : " + balaCacheWithPkgPath.toString()));
         }
 
-        createBalaFileDirectory(balaCacheWithPkgPath, logFormatter);
-        writeBalaFile(balaDownloadResponse, balaCacheWithPkgPath.resolve(balaFile), pkgOrg + "/" + pkgName +
+        // Create the following temp path bala/<org-name>/<pkg-name>/<pkg-version>/temp
+        Path tempPath = pkgPathInBalaCache.resolve(validPkgVersion).resolve("temp");
```

Comment:
In FileSystemRepository, we check for the availability of packages by version directory. There can be invalid balas captured as valid with this temp directory structure. Shall we take the temp directory as `bala/<org-name>/<pkg-name>/<pkg-version>temp` (e.g. ballerina/io/1.2.1temp) to avoid this?

Reply:
As suggested, i am now modifying the version folder with an appended temp. When the files are written to the platform directory, the version folder is renamed.